exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/CryptocurrencyServiceImpl.java

```diff
@@ -48,11 +50,30 @@ public void createPublicApiHandlers(Node node, Router router) {
 
   @Override
   @SuppressWarnings("ConstantConditions")
-  public HashCode submitTransaction(Transaction tx)
-      throws InvalidTransactionException, InternalServerError {
+  public HashCode submitTransaction(Transaction tx) {
     checkBlockchainInitialized();
-    node.submitTransaction(tx);
-    return tx.hash();
+    try {
+      node.submitTransaction(tx);
+      return tx.hash();
+    } catch (InvalidTransactionException | InternalServerError e) {
+      throw new RuntimeException("Propagated transaction submission exception", e);
+    }
+  }
+
+  @Override
+  @SuppressWarnings("ConstantConditions")
+  public Optional<Wallet> getWallet(PublicKey ownerKey) {
+    checkBlockchainInitialized();
+
+    return node.withSnapshot((view) -> {
+      CryptocurrencySchema schema = new CryptocurrencySchema(view);
+      MapIndex<PublicKey, Wallet> wallets = schema.wallets();
+      if (!wallets.containsKey(ownerKey)) {
```

Comment:
These three lines are redundant with `Optional.ofNullable` below — the MapIndex is specified to return null if there is no mapping for the given key.

Reply:
True! Pardon my obliviousness in this case.